### DISCUSSION TITLE
Scala like `apply` syntax

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#5120] [Ideas]
The idea is that `Ns(x, y, z)` is desugared into `Ns.apply(x, y, z)`.

Pros
* This seems like nice syntax for smart constructors, although less so with the lack of overloading.
* This could lead to a general version of expressions like `Set#{...}` and `Map#{...}`.

Cons
* It collides with the `Ns` tag of the outside scope, which could have conceptual friction and must be checked in the compiler.

**Comment by mlutze** [comment]
I don't think this works for Set and Map since we don't have variadic function arguments.

**Reply by JonathanStarup** [reply]
Yeah that would be the second part needed. With just the apply syntax, you could make `MyCollection(List#{a, b, c,})`
